robolectric/src/main/java/org/robolectric/android/internal/ParallelUniverse.java

```diff
@@ -106,6 +114,87 @@ public void setUpApplicationState(Method method, TestLifecycle testLifecycle, An
     ActivityThread activityThread = ReflectionHelpers.newInstance(ActivityThread.class);
     RuntimeEnvironment.setActivityThread(activityThread);
 
+    PackageParser.Package packageInfo = null;
+    if (appManifest.getAndroidManifestFile() != null && appManifest.getAndroidManifestFile().exists()) {
+
+      ResourcesManager resourcesManager = ResourcesManager.getInstance();
+      Resources resources;
+
+      if (RuntimeEnvironment.getApiLevel() == Build.VERSION_CODES.M) {
+        ReflectionHelpers.callInstanceMethod(ResourcesManager.class, resourcesManager, "applyConfigurationToResourcesLocked",
```

Comment:
Maybe these move to `ShadowResourcesManager.callApplyConfigurationToResourcesLocked()` (see `ShadowViewRootImpl`)?

Reply:
I've moved all this code into ShadowPackageParser now. I was able to use the public API on PackageParser (requires reflection on older SDKs so I figured to move it into a helper method.